src/rules/extensions.js

```diff
@@ -46,9 +46,10 @@ module.exports = {
       typeof configuration === 'object' ? configuration : context.options[1]
     )
 
-    function isUseOfExtensionRequired(extension) {
+    function isUseOfExtensionRequired(extension, isModule) {
       if (!has(modifiers, extension)) { modifiers[extension] = defaultConfig }
-      return modifiers[extension] === 'always'
+      const modifier = modifiers[extension]
+      return (!isModule && modifier === 'ignorePackages') || modifier === 'always'
```

Comment:
This regex is probably better understood if I split it:

```
function isMainImportOrScopedMainImport(importPath) {
  const mainImportRegExp = /^[\w]((?!\/).)*$/
  const scopedMainImportRegExp = /^@[^\/]+\/?[^\/]+$/
  return import.test(mainImportRegExp) || import.test(scopedMainImportRegExp)
}
```

This probably belongs in `importType.js`, and I might split it into 2 functions if I can think of the names for them :)